File: ts/worksheet/src/dressdiscover/api/models/worksheet/worksheet_feature_value_id.ts
export class WorksheetFeatureValueId {
    public static parse(id: string): WorksheetFeatureValueId {
        if (id.trim().length == 0) {
            throw new RangeError("id is empty");
        }
        return new WorksheetFeatureValueId(id);
    }

    public equals(other?: WorksheetFeatureValueId): boolean {
        if (typeof(other) === "undefined") {
            return false;
        }
        return this._id == other._id;
    }

    public toString(): string {
        return this._id;
    }

    private constructor(id: string) {
        this._id = id;
    }

    private _id: string;
}

File: cs/DressDiscover/DressDiscover.Api/src/DressDiscover/Api/Models/Worksheet/WorksheetFeatureValueId.cs
namespace DressDiscover.Api.Models.Worksheet
{
    public sealed class WorksheetFeatureValueId {
        public static WorksheetFeatureValueId Parse(string value) {
            if (value.Length == 0) {
                throw new System.ArgumentException("id is empty", "WorksheetFeatureValueId");
            }
            return new WorksheetFeatureValueId(value);
        }

        private WorksheetFeatureValueId(string value) {
            this.value = value;
        }

        public override string ToString() {
            return this.value;
        }

        private string value;
    }
}
